apps/omg_child_chain/lib/omg_child_chain/ethereum_event_aggregator.ex

```diff
@@ -0,0 +1,269 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+defmodule OMG.ChildChain.EthereumEventAggregator do
+  @moduledoc """
+  This process combines all plasma contract events we're interested in and does eth_getLogs + enriches them if needed
+  for all Ethereum Event Listener processes. 
+  """
+  use GenServer
+  require Logger
+
+  alias OMG.Eth.RootChain.Abi
+  alias OMG.Eth.RootChain.Event
+  alias OMG.Eth.RootChain.Rpc
+
+  @timeout 55_000
+
+  def deposit_created(server \\ __MODULE__, from_block, to_block) do
+    GenServer.call(server, {:deposit_created, from_block, to_block}, @timeout)
+  end
+
+  def in_flight_exit_started(server \\ __MODULE__, from_block, to_block) do
+    GenServer.call(server, {:in_flight_exit_started, from_block, to_block}, @timeout)
+  end
+
+  def in_flight_exit_piggybacked(server \\ __MODULE__, from_block, to_block) do
+    # input and output
+    GenServer.call(server, {:in_flight_exit_piggybacked, from_block, to_block}, @timeout)
+  end
+
+  def exit_started(server \\ __MODULE__, from_block, to_block) do
+    GenServer.call(server, {:exit_started, from_block, to_block}, @timeout)
+  end
+
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, opts, name: Keyword.get(opts, :name, __MODULE__))
+  end
+
+  def init(opts) do
+    contracts = opts |> Keyword.fetch!(:contracts) |> Map.values() |> Enum.map(&from_hex(&1))
+    # events = [[signature: "ExitStarted(address,uint160)", name: :exit_started, enrich: true],..]
+    events =
+      opts
+      |> Keyword.fetch!(:events)
+      |> Enum.map(&Keyword.fetch!(&1, :name))
+      |> Event.get_events()
+      |> Enum.zip(Keyword.fetch!(opts, :events))
+      |> Enum.reduce([], fn {signature, event}, acc -> [Keyword.put(event, :signature, signature) | acc] end)
+
+    events_signatures =
+      opts
+      |> Keyword.fetch!(:events)
+      |> Enum.map(&Keyword.fetch!(&1, :name))
+      |> Event.get_events()
+
+    ets_bucket = Keyword.fetch!(opts, :ets_bucket)
+    rpc = Keyword.get(opts, :rpc, Rpc)
+    # :ok = event_bus.subscribe("ethereum_new_height", link: true)
+
+    {:ok,
+     %{
+       # 200 blocks of events will be kept in memory
+       delete_events_threshold_height_blknum: 200,
+       ets_bucket: ets_bucket,
+       event_signatures: events_signatures,
+       events: events,
+       contracts: contracts,
+       rpc: rpc
+     }}
+  end
+
+  # I've  not quite convinced this is benefical for us in any way.
```

Comment:
sorry, I must've just missed it. As far as I understand what it is doing, _I think_ it can be got rid of.

Reply:
yeah, I actually left this (and the handle_info/2) in the code on purpose just so that PR reviewers can see I've tried to be more aggressive in syncing the events. but as said - there was no added value to this approach.